src/group_theory/subgroup/zpowers.lean

```diff
@@ -150,13 +147,45 @@ instance zpowers_is_commutative (g : G) : (zpowers g).is_commutative :=
 lemma zpowers_le {g : G} {H : subgroup G} : zpowers g ≤ H ↔ g ∈ H :=
 by rw [zpowers_eq_closure, closure_le, set.singleton_subset_iff, set_like.mem_coe]
 
+alias zpowers_le ↔ _ zpowers_le_of_mem
+alias add_subgroup.zmultiples_le ↔ _ _root_.add_subgroup.zmultiples_le_of_mem
+
+attribute [to_additive zmultiples_le_of_mem] zpowers_le_of_mem
+
 @[simp, to_additive zmultiples_eq_bot] lemma zpowers_eq_bot {g : G} : zpowers g = ⊥ ↔ g = 1 :=
 by rw [eq_bot_iff, zpowers_le, mem_bot]
 
+@[to_additive zmultiples_ne_bot] lemma zpowers_ne_bot : zpowers g ≠ ⊥ ↔ g ≠ 1 :=
+zpowers_eq_bot.not
+
 @[simp, to_additive zmultiples_zero_eq_bot] lemma zpowers_one_eq_bot :
    subgroup.zpowers (1 : G) = ⊥ :=
 subgroup.zpowers_eq_bot.mpr rfl
 
+@[to_additive coe_zmultiplies_subset] lemma coe_zpowers_subset (h_one : (1 : G) ∈ s)
+  (h_mul : ∀ a ∈ s, a * g ∈ s) (h_inv : ∀ a ∈ s, a * g⁻¹ ∈ s) : ↑(zpowers g) ⊆ s :=
+begin
+  rintro _ ⟨n, rfl⟩,
+  induction n using int.induction_on with n ih n ih,
+  { rwa zpow_zero },
+  { rw zpow_add_one,
+    exact h_mul _ ih },
+  { rw zpow_sub_one,
+    exact h_inv _ ih }
+end
```

Comment:
I'm not convinced by this lemma; the unbundled version seems more interesting:
```lean
on the right by `g` and `g⁻¹` on the right.  -/
@[to_additive zsmul_induction] lemma pow_induction_right {P : G → Prop} (h_one : P (1 : G))
  (h_mul : ∀ a, P a → P (a * g)) (h_inv : ∀ a, P a → P (a * g⁻¹)) : ∀ z : ℤ, P (g ^ z) :=
begin
  intro n,
  induction n using int.induction_on with n ih n ih,
  { rwa zpow_zero },
  { rw zpow_add_one,
    exact h_mul _ ih },
  { rw zpow_sub_one,
    exact h_inv _ ih }
end
```

Reply:
Perhaps combined with the lemma
```
@[to_additive coe_zmultiplies_subset] lemma coe_zpowers_subset :
  ↑(zpowers g) ⊆ s ↔ ∀ z : ℤ, g ^ z ∈ s :=
set.range_subset_iff
```

Reply:
Done. I haven't added `coe_zpowers_subset` because `forall_mem_zpowers` advantageously replaces it.